### ISSUE TITLE
EPIC: Model training, SHAP selection and backtesting on the GOLD marts (Phase 6)

### ISSUE THREAD

**Issue by Analyst-Ninja** [Analyst-Ninja/aurum#50]
## Problem / Motivation

The medallion warehouse is complete (EPIC #31, Phases 0–5). `gold.mart_training_set` holds **2,895,171 rows** across **503 symbols**, 2000-01-03 → 2026-09-02, **228 columns**, **321 monthly folds** — measured 2026-09-05. Nothing consumes it.

`src/modeling/` is a 0-byte `__init__.py`. `pyproject.toml` carries no ML dependency. `seeds/selected_features.csv` still holds its single placeholder row (`ret_21d,1,0.0,true,placeholder`). Two shipped dbt artifacts — `mart_feature_summary` and the SHAP loop it closes — are structurally waiting on a producer that does not exist.

This epic builds that producer: preprocessing, purged walk-forward training, SHAP-driven feature selection, a backtesting engine, a flat-file model registry, and a containerized training image.

Corresponds to **spec §3.7 (ML Training Pipeline)**, build **Phase 6**.

## What we want to do

A config-driven modeling subsystem under `src/modeling/`, mirroring the ingestion framework's ethos (one YAML fully specifies a run; nothing wired in Python):

```bash
uv run python -m src.modeling.cli train           -c src/modeling/configs/lgbm_xs_excess_5d.yaml
uv run python -m src.modeling.cli evaluate        -c ... --version 20260905-abc1234
uv run python -m src.modeling.cli select-features -c ... --version 20260905-abc1234
uv run python -m src.modeling.cli backtest        -c ... --version latest
uv run python -m src.modeling.cli predict         -c ... --version latest --asof 2026-09-05
```

**Design decisions already taken** (documented in `docs/modeling/`, see References):

| Decision | Choice |
|---|---|
| Primary target | `fwd_ret_5d_excess` — regression, market-neutral 5-day log excess return |
| Model | LightGBM, L2 on a per-date winsorized + standardized target |
| Validation | Purged + embargoed expanding walk-forward; annual refit; 24-month untouched holdout |
| Headline metric | ICIR (mean per-date Spearman IC / std × √252), plus decile spread and net long-short Sharpe |
| Feature selection | Per-fold TreeSHAP, collinearity-pruned, cumulative-95% cutoff capped at top-40 |
| Registry | Flat files under `models/{version}/`; MLflow deferred per spec §7 Q5 |
| Container | Training image only — no Airflow, no MLflow service, no serving |

## Implementation notes

Ordered; each child issue is a reviewable increment.

| Issue | Scope |
|---|---|
| #51 | Scaffolding — `modeling` dependency group, YAML config, Postgres loader, Parquet cache |
| #52 | Preprocessing + purged/embargoed walk-forward splits + manifests |
| #53 | LightGBM trainer with custom IC early-stopping + flat-file registry |
| #54 | Evaluation harness — IC/ICIR/decile spread/LS Sharpe, baselines, breakdowns |
| #55 | SHAP pipeline + the first real `selected_features.csv` |
| #56 | Backtesting engine — portfolio construction, costs, factor attribution, reality checks |
| #57 | Containerized training + re-enable pytest in CI |

#51 → #52 → #53 are strictly sequential. #54, #55, #56 all depend on #53 and can then proceed in parallel. #57 depends only on #51.

**The invariant this epic must not break.** `mart_features` carries no target column, by construction. Any preprocessing step that is not replayable against `mart_features` alone cannot be used at inference. `build_features()` is shared between training and inference, and the column order is pinned in `feature_manifest.json`.

**Known upward biases in every number this epic will produce**, to be restated in each report rather than assumed known:
- Survivorship — `company_meta.csv` is today's S&P 500 membership applied back to 2000.
- Point-in-time lag is `period_end + 60/90d`, not a real `filed_date` (#47).
- Overlapping 5-day labels reduce effective N roughly 5×.

## Acceptance criteria

- [ ] All seven child issues closed
- [ ] `uv run python -m src.modeling.cli train -c src/modeling/configs/lgbm_xs_excess_5d.yaml` produces a versioned artifact under `models/`
- [ ] `models/latest/metrics.json` reports ICIR, decile spread and long-short Sharpe against all four baselines on the 24-month holdout
- [ ] `models/latest/backtest/summary.json` reports net Sharpe by cost level, break-even bps, factor-attribution intercept and randomization p-value
- [ ] `seeds/selected_features.csv` holds a real SHAP ranking, and the narrowed model matched or beat the full-feature baseline on the holdout
- [ ] `dbt build --select mart_feature_summary` succeeds against the real seed
- [ ] Training runs inside the container against host Postgres
- [ ] `uv run pytest tests/ -v` passes and the pytest step is live in CI
- [ ] `uv run ruff check src/ main.py` clean

## References

- `docs/modeling/modeling-design.md` — target and model choice, metrics, limitations
- `docs/modeling/preprocessing-contract.md` — filters, deny-lists, the train/inference symmetry contract
- `docs/modeling/training-and-retraining.md` — folds, hyperparameters, registry, retraining policy
- `docs/modeling/feature-selection-shap.md` — the SHAP loop, producer side
- `docs/modeling/backtesting.md` — portfolio construction, costs, reality checks
- `docs/warehouse/rationale/gold-models-rationale.md` §5 — targets, folds, the leakage contract
- `docs/warehouse/rationale/selected-features-seed.md` — the seed, consumer side
- `docs/architecture/TECHNICAL_SPEC.md` §3.7, §6 Phase 6, §7 Q3/Q5
- EPIC #31 (the warehouse this consumes), #47 (real `filed_date`)
